src-tauri/src/algorithms/stoer_wagner.rs

```diff
@@ -24,29 +24,42 @@ pub fn st_mincut(g: &mut StoerWagnerGraph) -> Option<Cut> {
         }
         bheap.build_heap();
     }
-
-    let weight = a[a.len() - 1].weight;
     if a.len() < 2 {
         return None;
     }
+    let weight = a[a.len() - 1].weight;
 
     let s = a[a.len() - 1].node.name.clone();
     let t = a[a.len() - 2].node.name.clone();
 
     return Some(Cut::new(weight, s, t));
 }
 
-pub fn stoer_wagner(graph: &mut StoerWagnerGraph) -> Cut {
+pub fn stoer_wagner(graph: &mut StoerWagnerGraph) -> Option<Cut> {
```

Comment:
Why the changes?

Reply:
Thanks for asking! :) If there are no nodes in the graph, the app crashes because the cut returns None (see `stoer_wagner` function; in stoer_wagner.rs, line 61). I wrapped this both functions in an Option so we don't panic if we unwrap that value, and updated the unit test to take an Option instead of a Cut. 

So there are no changes to the return logic of `stoer_wagner` itself, just to the way we're handling its return value.

Reply:
Could also be handled with an `unwrap_or` with reasonable defaults, but I didn't think it was worth it to make up a cut (especially since we prioritize cuts by weight, so we'd need to set a high weight). That felt contrived, so I used `Option` instead.